logger.go

```diff
@@ -161,6 +164,27 @@ func (l *Logger) logf(format string, args ...interface{}) {
 		return
 	}
 
+	// if slog handler is set, use it
+	if l.slogHandler != nil {
+		record := slog.Record{
```

Comment:
Why not use NewRecord here as well? `slog.NewRecord(l.now(), stringToLevel(lv), msg, 0)`